packages/jmix-react-web/src/crud/editor/ui-callbacks/useSubmitCallback.ts

```diff
@@ -2,45 +2,49 @@ import {useCallback} from "react";
 import {
   addIdIfExistingEntity,
   EntityInstance,
-  findEntityMetadata,
-  GraphQLMutationFn,
   Metadata,
   toIdString,
   useMetadata,
-  unCapitalizeFirst,
-  MetaClassInfo
 } from "@haulmont/jmix-react-core";
-import { PersistEntityCallbacks, persistEntity } from "../util/persistEntity";
+import { ApolloError, FetchResult } from "@apollo/client";
+import {GraphQLError} from 'graphql';
 
-export interface SubmitCallbackHookOptions<TEntity, TData, TMutationVars> {
-  executeUpsertMutation: GraphQLMutationFn<TData, TMutationVars>;
-  updateResultName: string;
-  listQueryName: string;
+export interface SubmitCallbackCallbacks<TEntity, TData> {
+  onCommit?: (entityInstance: EntityInstance<TEntity>) => void;
+  onCreate?: (data?: TData | null) => void;
+  onEdit?: (data?: TData | null) => void;
+  onError?: (errors?: readonly GraphQLError[]) => void;
+  onApolloError?: (error: Error | ApolloError) => void;
+}
+
+export interface SubmitCallbackHookOptions<TEntity, TData> {
+  persistEntity: (updatedEntity: TEntity) => Promise<FetchResult<TData, Record<string, any>, Record<string, any>>>;
   entityName: string;
-  goToParentScreen: () => void;
   entityId?: string;
   entityInstance?: EntityInstance<TEntity>;
   onCommit?: (value: EntityInstance<TEntity>) => void;
+  onCreate?: (data?: TData | null) => void;
+  onEdit?: (data?: TData | null) => void;
+  onError?: (errors?: readonly GraphQLError[]) => void;
+  onApolloError?: (error: Error | ApolloError) => void;
   uiKit_to_jmixFront: (item: any, entityName: string, metadata: Metadata, stringIdName?: string) => Record<string, any>;
-  persistEntityCallbacks?: PersistEntityCallbacks
 }
 
 export function useSubmitCallback<
   TEntity = unknown,
-  TData extends Record<string, any> = Record<string, any>,
-  TMutationVars = unknown
+  TData extends Record<string, any> = Record<string, any>
 >({
-  executeUpsertMutation,
-  updateResultName,
-  listQueryName,
+  persistEntity,
```

Comment:
As discussed, it looks like `persistEntity` should be a default implementation for `onCommit`.